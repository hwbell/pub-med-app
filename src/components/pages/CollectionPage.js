import React from 'react';
import '../../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import Header from '../Header';
import ArticleResult from '../ArticleResult';
import EmailForm from "../EmailForm";
import Collection from '../Collection';
import { ButtonGroup, Button } from 'reactstrap';
import Loader from 'react-loader-spinner';

// animation
import posed, { PoseGroup } from 'react-pose';

// pose containers
const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

const initialState = {
  pdfReady: false,
  showPreview: false,
  emailModal: false,
  exportSelected: true,
  previewIndex: null
}

class CollectionPage extends React.Component {
  constructor(props) {
    super(props);

    this.renderLoader = this.renderLoader.bind(this);
    this.toggleEmailForm = this.toggleEmailForm.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleDelete = this.handleDelete.bind(this);

    this.state = initialState;

  }

  componentDidMount() {
    this.setState({
      pdfReady: true
    })
  }

  renderLoader() {
    return (
      <Div style={styles.loaderHolder} key="loader">
        <Loader
          height={100}
          width={100}
          type="ThreeDots"
          color="whitesmoke"
        />
      </Div>
    )
  }

  // for removing articles, registers the removal in App
  handleSubmit(article, collection) {
    console.log(`removing article: ${article.id} from user's ${collection.name} collection`);

    this.props.modifyCollection(article, collection.name, -1, () => {
      console.log('article removed from collection')
    })

  }

  // deletes the entire collection, registers the removal in App
  handleDelete(collection) {
    this.props.deleteCollection(collection, () => {
      console.log(`${collection.name} deleted from collecetions`)
    });
  }

  toggleEmailForm() {
    this.setState(prevState => ({
      emailModal: !prevState.emailModal
    }));
  }

  render() {

    console.log(JSON.parse(localStorage.getItem('user')))
    console.log(JSON.parse(localStorage.getItem('token')))


    return (
      <div className="search-page page">

        <div className="glass">

          {/* the header */}
          <Header
            class="heading"
            title={"PMC Collections"}
            subtitle={"create & share lists of resources"}
          />

          {/* results */}

          {/* the collections when they appear */}
          {this.props.collections.length > 0 ?

            this.props.collections.map((collection, i) => {

              return (
                <Collection 
                  key={i}
                  collection={collection}
                  handleSubmit={this.handleSubmit}  
                  handleDelete={this.handleDelete}
                  refreshUserCollections={this.props.refreshUserCollections}
                  />
              )

            })

            // or if there aren't any collections yet
            :
            <div className="outline" style={styles.content}>
              <p className="paragraph">
                {`It looks like you haven't made any new collections yet! You can add
                articles by searching the database, then organize, share & export them here. `}
              </p>
            </div>}




        </div>
      </div>
    );
  }
}

const styles = {
  main: {
    width: '100%'
  },
  titleHolder: {
    width: '100%',
    padding: '15px',
    borderRadius: '8px',
    backgroundColor: 'rgba(0, 0, 0, 0.24)'
  },
}

export default CollectionPage;
