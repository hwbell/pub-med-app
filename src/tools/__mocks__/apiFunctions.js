const fetch = require('node-fetch')

// query and return based on date / citations
export async function getArticles(query, sortParam) {

  // must have a query
  if (!query) return {};

  let allowedSorters = [
    '%20sort_cited:y',
    '%20sort_date:y',
  ];

  // check for bad strings and default to no sorter, which will sort by relevance
  let sorter;
  if (!sortParam) {
    sorter = '';
  } else {
    sorter = `%20sort_${sortParam}:y`;
    if (!allowedSorters.indexOf(sorter)) {
      sorter = '';
    }
  }

  // let searchUrl = `https://www.ebi.ac.uk/europepmc/webservices/rest/searchPOST?query=${query}${sorter}&resultType=core&pageSize=10&format=json`;

  // let headers = {}
  // headers['Content-Type'] = 'application/x-www-form-urlencoded';

  let list = [];
  for (let i = 0; i < 50; i++) {
    list.push({
      title: `Article ${i + 1}`,
      pmid: `PMID10000${i + 1}`
    })
  }

  let articles = await Promise.resolve({
    resultList: {
      result: list
    }
  })

  return articles;
}

// parses the result of a search query to a simple array of the article titles
export function parseSearchToTitlesArray(searchResults) {
  let titlesArray = searchResults.resultList.result.map((result) => {
    return result.title;
  })

  return titlesArray;
}