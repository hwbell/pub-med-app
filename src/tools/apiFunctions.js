const fetch = require('node-fetch')

// query and return based on date / citations
const getArticles = async (query, sortParam) => {

  // must have a query
  if (!query) return {};

  let allowedSorters = [
    '%20sort_cited:y',
    '%20sort_date:y',
  ];

  // check for bad strings and default to no sorter, which will sort by relevance
  let sorter;
  if (!sortParam) {
    sorter = '';
  } else {
    sorter = `%20sort_${sortParam}:y`;
    if (!allowedSorters.indexOf(sorter)) {
      sorter = '';
    }
  }

  let searchUrl = `https://www.ebi.ac.uk/europepmc/webservices/rest/searchPOST?query=${query}${sorter}&resultType=core&pageSize=10&format=json`;
  
  let headers = {}
  headers['Content-Type'] = 'application/x-www-form-urlencoded';

  let articles = await fetch(searchUrl, {
    method: 'POST',
    headers
  })
    .then(response => response.json())
    .then((response) => {
      // console.log(response)
      return response;
    })
    .catch(err => console.log(err))

  return articles;
}

// parses the result of a search query to a simple array of the article titles
const parseSearchToTitlesArray = (searchResults) => {
  let titlesArray = searchResults.resultList.result.map((result) => {
    return result.title;
  })
  return titlesArray;
}

// let results = getArticles('cancer').then((response) => {
//   console.log(response.resultList.result.slice(0,2))
// })

module.exports = {
  getArticles,
  parseSearchToTitlesArray,
}
