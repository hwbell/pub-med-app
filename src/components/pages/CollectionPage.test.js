import React from 'react';
import CollectionPage from './CollectionPage';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { getArticles } from '../../tools/apiFunctions.js';

describe('CollectionPage', () => {

  // stubs
  const modifyStub = jest.fn();
  let articles;
  let someProps;
  // get some props
  beforeAll(async () => {
    articles = await getArticles('medicine').then((response) => {
      return response.resultList.result;
    });

    someProps = {
      collections: [
        {
          name: 'first collection',
          articles: articles
        },
        {
          name: 'second collection',
          articles: articles
        }
      ],
      user: {
        name: 'Mark',
        email: 'mark@mail.com',
        age: 33,
        collections: [
          {
            name: 'first collection',
            articles: articles
          }
        ],
      },
      modifyCollection: modifyStub
    }
  });

  // tests
  it('renders without crashing', async () => {
    let wrapper = shallow(<CollectionPage {...someProps} />);
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<CollectionPage  {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements', () => {
    let wrapper = shallow(<CollectionPage  {...someProps} />);

    ['.page', '.glass'].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(1);
    });

    ['.collection-block', '.subtitle'].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(2);
    });


  });

  it('should change content when collections are empty', () => {
    let wrapper = shallow(<CollectionPage  {...someProps} />);

    // the explanations should not appear with collections present 
    ['.outline', '.paragraph'].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(0);
    });

    // set the user's collections to empty 
    wrapper.setProps({
      user: {
        name: 'Mark',
        email: 'mark@mail.com',
        age: 33,
        collections: []
      }
    });

    // now the 1st segment appears
    ['.outline', '.paragraph'].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(1);
    });

    //  set the new collections to empty 
    wrapper.setProps({
      collections: []
    });

    // now the second segment appears
    ['.outline', '.paragraph'].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(2);
    });

  })

});

