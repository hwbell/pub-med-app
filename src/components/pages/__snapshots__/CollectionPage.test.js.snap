// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage renders correctly 1`] = `
<div
  className="search-page page"
>
  <div
    className="glass"
  >
    <div
      className="heading"
    >
      <h2
        className="title"
      >
        PMC Collections
      </h2>
      <h4
        className="subtitle"
      >
        create & share lists of resources
      </h4>
    </div>
    <div
      className="outline"
    >
      <div
        className="row"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.24)",
            "borderRadius": "8px",
            "padding": "15px",
            "width": "100%",
          }
        }
      >
        <div
          className="row"
        >
          <p
            className="collection-title"
          >
            <strong>
              first collection 
            </strong>
             [ 10 ]
          </p>
        </div>
        <div
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          <button
            aria-label={null}
            className="add article-button btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            preview
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Improving access to Indigenous medicine for patients in hospital-based settings: a challenge for health systems in northern Canada.
          </p>
          <p
            className="article-title"
          >
            Redvers N, Marianayagam J, Blondin B.
          </p>
          <p
            className="article-title"
          >
            International journal of circumpolar health  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Serotype Distribution and Antimicrobial Resistance of Invasive and Noninvasive Streptococcus pneumoniae Isolates in Korea between 2014 and 2016.
          </p>
          <p
            className="article-title"
          >
            Park DC, Kim SH, Yong D, Suh IB, Kim YR, Yi J, Song W, Song SA, Moon HW, Lee HK, Park KU, Kim S, Jeong SH, Lee J, Jeong J, Kim YK, Lee M, Cho J, Kim JW, Shin KS, Hwang SH, Chung JW, Woo HI, Lee CH, Ryoo N, Chang CL, Kim HS, Kim J, Shin JH, Kim SH, Lee MK, Lee SG, Jang SJ, Lee K, Suh H, Sohn YH, Kwon MJ, Lee HJ, Hong KH, Woo KS, Park CM, Shin JH.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Long-term safety and efficacy of deferasirox in patients with myelodysplastic syndrome, aplastic anemia and other rare anemia in Taiwan.
          </p>
          <p
            className="article-title"
          >
            Ko BS, Chang MC, Chiou TJ, Chang TK, Chen YC, Lin SF, Chang CS, Lu YC, Yeh SP, Chen TY, Hwang WS.
          </p>
          <p
            className="article-title"
          >
            Hematology (Amsterdam, Netherlands)  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Culinary medicine and community partnership: hands-on culinary skills training to empower medical students to provide patient-centered nutrition education. 
          </p>
          <p
            className="article-title"
          >
            Pang B, Memel Z, Diamant C, Clarke E, Chou S, Gregory H.
          </p>
          <p
            className="article-title"
          >
            Medical education online  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            First Case in Korea of a Patient With Anti-PP1Pk Antibodies: Successful Blood Management via Acute Normovolemic Hemodilution.
          </p>
          <p
            className="article-title"
          >
            Ha C, Choi S, Yu H, Chun S, Kim KH, Lee JH, Han IW, Cho D.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            A 360-degree assessment of teaching effectiveness using a structured-videorecorded observed teaching exercise for faculty development.
          </p>
          <p
            className="article-title"
          >
            Jones CA, Watkins FS, Williams J, Lambros A, Callahan KE, Lawlor J, Williamson JD, High KP, Atkinson HH.
          </p>
          <p
            className="article-title"
          >
            Medical education online  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Development of Statistical Software for the Korean Laboratory Accreditation Program Using R Language: LaboStats.
          </p>
          <p
            className="article-title"
          >
            Kim Y, Park HI, Lee HK, Nam HS, Lee YW, Lee SG, Park Y, Lee W, Nam MH, Song SH, Chung JW, Lee J.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Comparison of Six Commercial Diagnostic Tests for the Detection of Dengue Virus Non-Structural-1 Antigen and IgM/IgG Antibodies.
          </p>
          <p
            className="article-title"
          >
            Lee H, Ryu JH, Park HS, Park KH, Bae H, Yun S, Choi AR, Cho SY, Park C, Lee DG, Lim J, Lee J, Lee S, Shin S, Park H, Oh EJ.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Improvement in External Quality Assessment Results for Qualitative Fecal Immunochemical Tests in Korea After Feedback to Manufacturers.
          </p>
          <p
            className="article-title"
          >
            Lee W, Ryoo N, Suh HS, Jeon CH, Urinalysis and Routine Microscopy Subcommittee, Korean Association of External Quality Assessment Service.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            First Isolation of Mycobacterium virginiense From a Human Pulmonary Specimen.
          </p>
          <p
            className="article-title"
          >
            Jung J, Yoo IY, Jhun BW, Koh WJ, Lee NY, Huh HJ.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
    </div>
    <div
      className="outline"
    >
      <div
        className="row"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.24)",
            "borderRadius": "8px",
            "padding": "15px",
            "width": "100%",
          }
        }
      >
        <div
          className="row"
        >
          <p
            className="collection-title"
          >
            <strong>
              second collection 
            </strong>
             [ 10 ]
          </p>
        </div>
        <div
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          <button
            aria-label={null}
            className="add article-button btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            preview
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Improving access to Indigenous medicine for patients in hospital-based settings: a challenge for health systems in northern Canada.
          </p>
          <p
            className="article-title"
          >
            Redvers N, Marianayagam J, Blondin B.
          </p>
          <p
            className="article-title"
          >
            International journal of circumpolar health  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Serotype Distribution and Antimicrobial Resistance of Invasive and Noninvasive Streptococcus pneumoniae Isolates in Korea between 2014 and 2016.
          </p>
          <p
            className="article-title"
          >
            Park DC, Kim SH, Yong D, Suh IB, Kim YR, Yi J, Song W, Song SA, Moon HW, Lee HK, Park KU, Kim S, Jeong SH, Lee J, Jeong J, Kim YK, Lee M, Cho J, Kim JW, Shin KS, Hwang SH, Chung JW, Woo HI, Lee CH, Ryoo N, Chang CL, Kim HS, Kim J, Shin JH, Kim SH, Lee MK, Lee SG, Jang SJ, Lee K, Suh H, Sohn YH, Kwon MJ, Lee HJ, Hong KH, Woo KS, Park CM, Shin JH.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Long-term safety and efficacy of deferasirox in patients with myelodysplastic syndrome, aplastic anemia and other rare anemia in Taiwan.
          </p>
          <p
            className="article-title"
          >
            Ko BS, Chang MC, Chiou TJ, Chang TK, Chen YC, Lin SF, Chang CS, Lu YC, Yeh SP, Chen TY, Hwang WS.
          </p>
          <p
            className="article-title"
          >
            Hematology (Amsterdam, Netherlands)  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Culinary medicine and community partnership: hands-on culinary skills training to empower medical students to provide patient-centered nutrition education. 
          </p>
          <p
            className="article-title"
          >
            Pang B, Memel Z, Diamant C, Clarke E, Chou S, Gregory H.
          </p>
          <p
            className="article-title"
          >
            Medical education online  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            First Case in Korea of a Patient With Anti-PP1Pk Antibodies: Successful Blood Management via Acute Normovolemic Hemodilution.
          </p>
          <p
            className="article-title"
          >
            Ha C, Choi S, Yu H, Chun S, Kim KH, Lee JH, Han IW, Cho D.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            A 360-degree assessment of teaching effectiveness using a structured-videorecorded observed teaching exercise for faculty development.
          </p>
          <p
            className="article-title"
          >
            Jones CA, Watkins FS, Williams J, Lambros A, Callahan KE, Lawlor J, Williamson JD, High KP, Atkinson HH.
          </p>
          <p
            className="article-title"
          >
            Medical education online  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Development of Statistical Software for the Korean Laboratory Accreditation Program Using R Language: LaboStats.
          </p>
          <p
            className="article-title"
          >
            Kim Y, Park HI, Lee HK, Nam HS, Lee YW, Lee SG, Park Y, Lee W, Nam MH, Song SH, Chung JW, Lee J.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Comparison of Six Commercial Diagnostic Tests for the Detection of Dengue Virus Non-Structural-1 Antigen and IgM/IgG Antibodies.
          </p>
          <p
            className="article-title"
          >
            Lee H, Ryu JH, Park HS, Park KH, Bae H, Yun S, Choi AR, Cho SY, Park C, Lee DG, Lim J, Lee J, Lee S, Shin S, Park H, Oh EJ.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            Improvement in External Quality Assessment Results for Qualitative Fecal Immunochemical Tests in Korea After Feedback to Manufacturers.
          </p>
          <p
            className="article-title"
          >
            Lee W, Ryoo N, Suh HS, Jeon CH, Urinalysis and Routine Microscopy Subcommittee, Korean Association of External Quality Assessment Service.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
      <div
        className="article-result"
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <p
            className="article-title"
          >
            First Isolation of Mycobacterium virginiense From a Human Pulmonary Specimen.
          </p>
          <p
            className="article-title"
          >
            Jung J, Yoo IY, Jhun BW, Koh WJ, Lee NY, Huh HJ.
          </p>
          <p
            className="article-title"
          >
            Annals of laboratory medicine  
            <strong>
                2019
            </strong>
          </p>
        </div>
        <div
          style={
            Object {
              "alignSelf": "flex-start",
              "marginLeft": "15px",
            }
          }
        >
          <button
            aria-label={null}
            className="article-button view btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            view article
          </button>
          <button
            aria-label={null}
            className="article-button warn btn btn-secondary btn-sm"
            onClick={[Function]}
            type="button"
          >
            remove
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
